content/en/docs/languages/net/instrumentation.md

```diff
@@ -63,320 +281,388 @@ using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    }
 
-If you have a distinct sub-operation you'd like to track as a part of another
-one, you can create activities to represent the relationship.
+    public ActivitySource ActivitySource { get; }
 
-```csharp
-public static void ParentOperation()
-{
-    using var parentActivity = MyActivitySource.StartActivity("ParentActivity");
+    public void Dispose()
+    {
+        this.ActivitySource.Dispose();
+    }
+}
+```
 
-    // Do some work tracked by parentActivity
+Then we will update the `Program.cs` to add the Instrument object as a dependency injection:
 
-    ChildOperation();
+```csharp 
+//...
 
-    // Finish up work tracked by parentActivity again
-}
+// Register the ActivitySource as a singleton in the DI container.
+// Create a service to expose the ActivitySource Instruments.
+builder.Services.AddSingleton<Instrumentation>();
 
-public static void ChildOperation()
-{
-    using var childActivity = MyActivitySource.StartActivity("ChildActivity");
+builder.Services.AddControllers();
 
-    // Track work in ChildOperation with childActivity
-}
-```
+var app = builder.Build();
 
-When you view spans in a trace visualization tool, `ChildActivity` will be
-tracked as a nested operation under `ParentActivity`.
+app.MapControllers();
 
-### Nested Activities in the same scope
+app.Run();
 
-You may wish to create a parent-child relationship in the same scope. Although
-possible, this is generally not recommended because you need to be careful to
-end any nested `Activity` when you expect it to end.
+```
+
+In the application file `DiceController.cs` we then reference that activitySource instance:
 
 ```csharp
-public static void DoWork()
+using Microsoft.AspNetCore.Mvc;
+using System.Diagnostics;
+using System.Net;
+
+public class DiceController : ControllerBase
 {
-    using var parentActivity = MyActivitySource.StartActivity("ParentActivity");
+    private ILogger<DiceController> logger;
 
-    // Do some work tracked by parentActivity
+    private ActivitySource activitySource;
 
-    using (var childActivity = MyActivitySource.StartActivity("ChildActivity"))
+     public DiceController(ILogger<DiceController> logger, Instrumentation instrumentation)
     {
-        // Do some "child" work in the same function
+        this.logger = logger;
+        this.activitySource = instrumentation.ActivitySource;
     }
 
-    // Finish up work tracked by parentActivity again
+    //...
 }
-```
 
-In the preceding example, `childActivity` is ended because the scope of the
-`using` block is explicitly defined, rather than scoped to `DoWork` itself like
-`parentActivity`.
 
-### Creating independent Activities
+```
 
-The previous examples showed how to create Activities that follow a nested
-hierarchy. In some cases, you'll want to create independent Activities that are
-siblings of the same root rather than being nested.
+The same activitySource is also passed to the library file `Dice.cs`:
```

Comment:
I think the code to do this is missing? You need to add something like new Dice(activitySource,min,max).